### PR TITLE
CSHARP-5021: Replace use of DisposableEnvironmentVariable with EnvironmentVariableProvider

### PR DESCRIPTION
**Pull request by adelinowona**

### CODE REVIEW

File: tests/MongoDB.Driver.Core.Tests/Core/Connections/ClientDocumentHelperTests.cs [hunk 1]
```diff
@@ -146,30 +146,30 @@ public void CreateEnvDocument_should_return_expected_result(bool isDockerToBeDet
 
             if (isKubernetesToBeDetected)
             {
-                environmentVariableProviderMock.Setup(p => p.GetEnvironmentVariable("KUBERNETES_SERVICE_HOST")).Returns("dummy");
+                environmentVariableProviderMock.Setup(env => env.GetEnvironmentVariable("KUBERNETES_SERVICE_HOST")).Returns("dummy");
             }
             else
             {
-                environmentVariableProviderMock.Setup(p => p.GetEnvironmentVariable("KUBERNETES_SERVICE_HOST")).Returns(null as string);
+                environmentVariableProviderMock.Setup(env => env.GetEnvironmentVariable("KUBERNETES_SERVICE_HOST")).Returns(null as string);
```
Comment (BorisDog) [inline tests/MongoDB.Driver.Core.Tests/Core/Connections/ClientDocumentHelperTests.cs:153]: Maybe shorter?
`environmentVariableProviderMock.Setup(env => env.GetEnvironmentVariable("KUBERNETES_SERVICE_HOST")).Returns(isKubernetesToBeDetected ? "dummy" : null);` ?
And 
`fileSystemProviderMock.Setup(fileSystem => fileSystem.File.Exists("/.dockerenv")).Returns(isDockerToBeDetected);`

**Review by BorisDog — APPROVED** [review]
LGTM + minor comment.
Nice cleanup.

File: tests/MongoDB.Driver.Core.Tests/Core/Connections/ClientDocumentHelperTests.cs (hunk 1 above)
Comment (adelinowona) [reply tests/MongoDB.Driver.Core.Tests/Core/Connections/ClientDocumentHelperTests.cs]: Done. Thanks!

**Review by BorisDog — APPROVED** [review]
LGTM!

**Pull request merged**